brian2/stateupdaters/base.py

```diff
@@ -127,10 +138,16 @@ def apply_stateupdater(equations, variables, method):
             if the_method is None:
                 raise ValueError(('No stateupdater that is suitable for the '
                                   'given equations has been found.'))
-            logger.info('Using state updater: %r' % the_method)
+            if group_name is not None:
+                msg_text = ("No numerical integration method specified for group "
+                            "'{group_name}', choosing method '{method}'.")
+            else:
+                msg_text = ("No numerical integration method specified, "
+                            "choosing method '{method}'.")
+            logger.info(msg_text.format(group_name=group_name,
+                                        method=the_method), 'method_choice')
```

Comment:
I'd vote for keeping it. After all, "exact" (which is called "linear", isn't it?) is problematic quite often. If you notice that the preparation time is taking forever and you see that it chooses the linear method after a long wait in the debug messages you know that it might be a good idea to explicitly choose a method.
But maybe the debug output should be rather like this:
```
DEBUG No numerical integration method specified for group 'neurongroup', trying the methods 'linear', 'exponential_euler', 'euler', 'heun' in order until one method can be applied to the equations ...
DEBUG ... selecting method 'linear'
```